src/main.js

```diff
@@ -145,6 +147,35 @@ class PercyClient {
     return this._httpPost(`${this.apiUrl}/builds/${buildId}/resources/`, data);
   }
 
+  uploadResources(buildId, contents) {
+    const _this = this;
+    function* generatePromises() {
+      for (const content of contents) {
+        yield _this.uploadResource(buildId, content);
+      }
+    }
+
+    const concurrency = 2;
+    const pool = new PromisePool(generatePromises(), concurrency);
```

Comment:
Using `es6-promise-pool` here as that's what I know you guys use in `ember-percy` and is also what I used in `react-percy`. That does mean this needs `regenerator-runtime` to properly support generators, though.